src/sdk/namespace_fs.js

```diff
@@ -602,27 +606,33 @@ class NamespaceFS {
                 if (drain_promise) {
                     await drain_promise;
                     drain_promise = null;
+                    object_sdk.throw_if_aborted();
                 }
 
                 // write the data out to response
                 buffer_pool_cleanup = null; // cleanup is now in the socket responsibility
                 const write_ok = res.write(data, null, callback);
                 if (!write_ok) {
-                    drain_promise = stream_utils.wait_drain(res);
+                    drain_promise = stream_utils.wait_drain(res, object_sdk.abort_controller.signal);
```

Comment:
on second thought, we better make wait_drain follow the same api as wait_finished and send an options object with a signal property...